### PR TITLE
change form lang

### PR DESCRIPTION
**Pull request by calellowitz**
https://dimagi-dev.atlassian.net/browse/USH-162

##### SUMMARY

This allows users to change  the language of a form in webapps while in that form, maintaining the current form session. Easiest to review commit by commit although it ended up pretty small.
Screen Shot 2020-09-01 at 11 12 49 AM

### CODE REVIEW

**Comment by calellowitz** [comment]
This depends on the formplayer PR mentioned above

**Comment by calellowitz** [comment]
Rebased this onto marionette 4 upgrade so that it would not conflict with the upgrade. That means it is not safe to merge until that PR is merged and only the last 3 commits (mine) should be reviewed.

File: corehq/apps/cloudcare/templates/formplayer/menu_list.html
Comment (orangejenny) [inline corehq/apps/cloudcare/templates/formplayer/menu_list.html:130]: Glyphicons are built into bootstrap, but HQ uses a different icon library, [fontawesome](https://fontawesome.com/v4.7.0/cheatsheet/). I don't see a three-dot icon, `fa-bars` is probably the closest. Something like `fa-chevron-down` or `fa-caret-down` would work, too.

File: corehq/apps/cloudcare/templates/formplayer/menu_list.html
Comment (orangejenny) [inline corehq/apps/cloudcare/templates/formplayer/menu_list.html:132]: `trans`

File: corehq/apps/cloudcare/static/cloudcare/js/formplayer/menus/controller.js
Comment (orangejenny) [inline corehq/apps/cloudcare/static/cloudcare/js/formplayer/menus/controller.js:91]: Should this check `menuResponse.langa.length > 1`?

File: corehq/apps/cloudcare/static/cloudcare/js/formplayer/menus/util.js
Comment (orangejenny) [inline corehq/apps/cloudcare/static/cloudcare/js/formplayer/menus/util.js:85]: Is there a reason most identifiers are named with "language" but a few use "locale"?

File: corehq/apps/cloudcare/templates/formplayer/menu_list.html
Comment (orangejenny) [inline corehq/apps/cloudcare/templates/formplayer/menu_list.html:137]: Not a must-have, but it'd be nice to show which language is currently selected.

**Comment by orangejenny** [comment]
Test failures look like they're mine, I'll get them fixed.

File: corehq/apps/cloudcare/static/cloudcare/js/formplayer/menus/util.js
Comment (calellowitz) [reply corehq/apps/cloudcare/static/cloudcare/js/formplayer/menus/util.js:85]: Formplayer code that dealt with this mostly called it a locale (although it does refer to langs in some places) and existing web apps code was inconsistent. I started using locale in most places, but some langs slipped through. In particular, in this case, I was using the `locales` (which is what we call them at the app level) property of the menuResponse, before I realized that when it corresponds to a form there is a separate property called `langs` that was a little nicer (did not include a "default"). The localeModels is probably a holdover from when I was iterating over locales rather than langs. Happy to make this slightly more consistent, although not certain which one is better.

File: corehq/apps/cloudcare/static/cloudcare/js/formplayer/menus/controller.js
Comment (calellowitz) [reply corehq/apps/cloudcare/static/cloudcare/js/formplayer/menus/controller.js:91]: yeah, I can do that.

**Review by orangejenny — APPROVED** [review]

File: corehq/apps/cloudcare/static/cloudcare/js/formplayer/menus/util.js
Comment (orangejenny) [reply corehq/apps/cloudcare/static/cloudcare/js/formplayer/menus/util.js:85]: Consistency within formplayer would be good, consistency across HQ would be great. My bias is to favor langs because that's what app manager uses, and that's unlikely to change (since couch attributes are a pain to rename). But I leave it to your judgment how much, if anything, to change in this PR.

File: corehq/apps/hqwebapp/static/cloudcare/less/formplayer-common/breadcrumbs.less
Comment (orangejenny) [inline corehq/apps/hqwebapp/static/cloudcare/less/formplayer-common/breadcrumbs.less:36]: Alternatively, there's a `clickable` class you can add in the HTML template.

**Comment by orangejenny** [comment]
Test failures are fixed in the parent branch, so if you merge `jls/marionette-3` travis should pass and the lint errors should decrease.

**Review by orangejenny — APPROVED** [review]

**Pull request merged**
